src/ocaml_flags.ml

```diff
@@ -5,47 +5,48 @@ open Build.O
 let default_ocamlc_flags   = ["-g"]
 let default_ocamlopt_flags = ["-g"]
 
-let dev_mode_warnings ~dune_version =
-  let codes =
-    [ 4
-    ; 29
-    ; 40
-    ; 41
-    ; 42
-    ; 44
-    ; 45
-    ; 48
-    ; 58
-    ; 59
-    ; 60
-    ]
-  in
-  let codes =
-    if dune_version >= (1, 11) then
-      codes @ [66]
-    else
-      codes
+let dev_mode_warnings =
+  (* New warnings should be introduced here *)
+  let all =
+    Int.Set.diff
+      (Int.Set.of_list (List.init 62 ~f:succ))
+      (Int.Set.of_list
+         [ 4
+         ; 29
+         ; 40
+         ; 41
+         ; 42
+         ; 44
+         ; 45
+         ; 48
+         ; 58
+         ; 59
+         ; 60
+         ]) in
+  let by_version =
+    match Syntax.Version.Per.of_mapping [] ~default:all with
```

Comment:
I'm not sure `Per_item` will work best here since we want to map intervals rather than individual keys. Using a bunch if `if` statements seems enough for this case.